### PR TITLE
Auctions per user

### PR DESCRIPTION
**Pull request by RobertMrowiec**
# Tasks:
- create myauction list,
- allow user to edit his auction

# Testing scenario:
1. login
2. add auction
3. click my auctions from navbar
4. check how it works / edit your created auction

# Preview
image

### CODE REVIEW

File: app/components/add-auction.js [hunk 1]
```diff
@@ -1,59 +1,63 @@
...
 		minimalPrice: 100,
-		name: 'test'
+		name: 'test title'
 	},
 
 	actions: {
 		async addAuction() {
-			
-			const carId = await this.store.createRecord('car', this.car).save().then((x) => x.id)
+			const actualUser = await this.currentUser.get()
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: Maybe we can create property `user` which will invoke `.get()` inside

File: app/components/add-auction.js [hunk 2]
```diff
@@ -1,59 +1,63 @@
...
+			const user = await this.store.findRecord('user', actualUser.id)
+			const car = await this.store.createRecord('car', this.car).save()
 
-			return this.store.createRecord('auction', { ...this.auction, car: carId })
+			return this.store.createRecord('auction', { ...this.auction, car, user})
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: keep consistency with empty space after opening curly braces and before the closing one

File: app/components/add-auction.js [hunk 3]
```diff
@@ -1,59 +1,63 @@
...
-		},
-
-		_uploadPhoto: task(function* (file) {
-			let	response = yield file.upload('http://localhost:8007/api/images')
-			set(this.auction, 'image', response.body)
-		})
+		}
 	},
 
+	_uploadPhoto: task(function* (file) {
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: 👍

Comment (RobertMrowiec) [reply app/components/add-auction.js]: It's not mine function :c

File: app/components/edit-auction.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+	car: null,
+	carId: null,
+	auction: null,
+	auctionId: null,
+
+	async init() {
+		this.set('auctionId', this.router.currentRoute.attributes.auctionId)
+		this._super(...arguments)
+		await this.store.findRecord('auction', this.auctionId, { include: 'car' }).then((auction) => {
```
Comment (Rxbsxn) [inline app/components/edit-auction.js]: I'd move the fetching of the record here to a separated method and use it in computed

File: app/components/add-auction.js (hunk 1 above)
Comment (RobertMrowiec) [reply app/components/add-auction.js]: But I think it needs to be ember data object, not just user from currentUser service

File: app/components/edit-auction.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	},
+
+	actions: {
+		async update() {
+			this.store.findRecord('auction', this.auctionId).then((auction) => {
+				auction.set('end-date', new Date(this.auction.endDate).toISOString())
+				auction.save()
+			})
+			this.store.findRecord('car', this.carId).then((car) => car.save())
+			this._correctUpdate()
```
Comment (Rxbsxn) [inline app/components/edit-auction.js]: Maybe `auctionUpdateNotification` or something like this, because `_correctUpdate` is not so meaningful

Comment (Rxbsxn) [reply app/components/edit-auction.js]: or `successNotification`

File: app/components/add-auction.js [hunk 6]
```diff
@@ -1,59 +1,63 @@
+import { get, set } from '@ember/object'
+
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: 💇

File: app/components/add-auction.js [hunk 7]
```diff
@@ -19,11 +19,11 @@ export default Component.extend({
 		model: 'car model',
 		horsePower: 10,
 		maxTorque: 199,
-		assembledAt: ''
+		assembledAt: (new Date()).toISOString().substring(0, 10)
```
Comment (Rxbsxn) [inline app/components/add-auction.js]: Why do you need these parentheses around `new Date()`

File: app/components/edit-auction.js [hunk 8]
```diff
@@ -25,19 +25,21 @@ export default Component.extend({
 
 	init() {
 		this._super(...arguments)
-		this.set('auctionId', this.router.currentRoute.attributes.auctionId)
+		this.set('auctionId', this.router.currentRoute ? this.router.currentRoute.attributes.auctionId : this.auctionId)
```
Comment (Rxbsxn) [inline app/components/edit-auction.js]: destruct { currentRoute }

File: app/components/edit-auction.js [hunk 9]
```diff
@@ -25,19 +25,21 @@ export default Component.extend({
...
-			this.store.findRecord('auction', this.auctionId).then((auction) => {
-				auction.set('end-date', new Date(this.auction.endDate).toISOString())
-				auction.save()
-			})
+			const auction = await this.store.findRecord('auction', this.auctionId)
+			const car = await this.store.findRecord('car', await this.car.get('id'))
 
-			this.store.findRecord('car', await this.car.get('id')).then((car) => car.save())
-			this._successNotification()
+			auction.set('end-date', new Date(this.auction.endDate).toISOString())
```
Comment (Rxbsxn) [inline app/components/edit-auction.js]: Why `end-date`? Please camelCase

File: app/styles/components/auction-list.sass [hunk 10]
```diff
@@ -1,9 +1,13 @@
 .auction-list
 	width: 100%
-	height: 90vh
-	& > div:first-child
-		margin: 0px 30px
+	height: 85vh
+	// & > div:first-child
+		// margin: 0px 30px
```
Comment (Rxbsxn) [inline app/styles/components/auction-list.sass]: 💇‍♂

File: app/styles/components/auction-list.sass [hunk 11]
```diff
@@ -1,9 +1,13 @@
...
 
 .auction-card
-	margin: 50px 30px 0 30px // I will change to padding somehow
+	// margin: 50px 30px 0 30px // I will change to padding somehow
```
Comment (Rxbsxn) [inline app/styles/components/auction-list.sass]: 💇

File: app/templates/auction-list.hbs [hunk 12]
```diff
@@ -1,10 +1,13 @@
-<div style='height: 15vh' class='d-flex align-items-center flex-column'>
+<div style='height: 20vh' class='d-flex align-items-center flex-column'>
```
Comment (Rxbsxn) [inline app/templates/auction-list.hbs]: Move inline styles to a dedicated class

File: app/templates/components/add-auction.hbs [hunk 13]
```diff
@@ -1,4 +1,4 @@
-<div class="add-auction-card bg-white d-flex w-75 row rounded-lg align-self-center position-relative" style='height: 75vh'>
+<div class="add-auction-card bg-white d-flex row rounded-lg align-self-center position-relative" style='height: 75vh'>
```
Comment (Rxbsxn) [inline app/templates/components/add-auction.hbs]: same here

File: app/templates/auction-list.hbs [hunk 14]
```diff
@@ -1,10 +1,13 @@
...
 	<h1 class='text-light'>Do You want to sell a car? Click the button below</h1>
 	<div>
 		{{#link-to 'addAuction'}}
 			<button type="button" class="btn btn-secondary">Click here</button>
 		{{/link-to}}
 	</div>
+	<div class="form-group">
+    <input type="text" class="form-control" placeholder="Search" style="margin-top: 30px">
```
Comment (Rxbsxn) [inline app/templates/auction-list.hbs]: same here

File: app/templates/components/auctions.hbs [hunk 15]
```diff
@@ -2,7 +2,7 @@
 
 	{{#each auctions as |auction|}}
 
-		<div class="card vw-25">
+		<div class="card vw-25" style='max-width: 450px'>
```
Comment (Rxbsxn) [inline app/templates/components/auctions.hbs]: inline styles -> class in external file

File: app/templates/components/auctions.hbs [hunk 16]
```diff
@@ -2,7 +2,7 @@
...
 			<img src={{auction.image}} class="card-img-top" alt="URL" style='height: 25vh'>
```
Comment (Rxbsxn) [inline app/templates/components/auctions.hbs]: same here

File: app/templates/components/edit-auction.hbs [hunk 17]
```diff
@@ -1,4 +1,4 @@
-<div class="add-auction-card bg-white d-flex w-75 row rounded-lg align-self-center position-relative" style='height: 75vh'>
+<div class="add-auction-card bg-white d-flex row rounded-lg align-self-center position-relative" style='height: 75vh'>
```
Comment (Rxbsxn) [inline app/templates/components/edit-auction.hbs]: ⛔️ inline

File: app/templates/components/edit-auction.hbs [hunk 18]
```diff
@@ -1,4 +1,4 @@
...
 	<h1 class='add-auction-card_title'> Edit auction </h1>
 		<div class="add-auction-card_left w-75">
 			<div class="w-100 px-5" style='padding-top: 10vh'>
```
Comment (Rxbsxn) [inline app/templates/components/edit-auction.hbs]: 🛑 inline

File: app/templates/components/edit-auction.hbs [hunk 19]
```diff
@@ -115,7 +115,7 @@
 			/>
 
 			<button
-				type="button" 
+				type="button"
 				data-test-auction-button
 				class="btn btn-outline-primary profile-button"
 				style='width: 10%'
```
Comment (Rxbsxn) [inline app/templates/components/edit-auction.hbs]: inline

File: app/components/add-auction.js (hunk 7 above)
Comment (RobertMrowiec) [reply app/components/add-auction.js]: I thought I need it to use .toISOSTring() later, but I see it's not necessary

File: app/components/edit-auction.js (hunk 9 above)
Comment (RobertMrowiec) [reply app/components/edit-auction.js]: json-api standard default make it with dash 👎

**Pull request merged**